packages/peregrine/lib/talons/ProductFullDetail/useProductFullDetail.js

```diff
@@ -221,6 +229,41 @@ export const useProductFullDetail = props => {
         [product, optionCodes, optionSelections]
     );
 
+    // The map of ids to values (and their uids)
+    // For example:
+    // { "179" => [{ uid: "abc", value_index: 1 }, { uid: "def", value_index: 2 }]}
+    const attributeIdToValuesMap = useMemo(
+        () =>
+            new Map(
+                product.configurable_options.map(option => [
+                    option.attribute_id,
+                    option.values
+                ])
+            ),
```

Comment:
Good use of `useMemo` to convert this array to a map. 👍 

Optionally, a small optimization would be to skip the intermediate array (faster, less memory):

```es6
const attributeIdToValuesMap = useMemo(
  () => {
    const map = new Map()

    for (const { attribute_id, values } of options) {
      map.set(attribute_id, values)
    }
  },
  [options]
)
```

Reply:
Done.